### DISCUSSION TITLE
Unreal Engine 5.2: Aircraft has bizzare behaviour when spawned too far from Level Origin

### DISCUSSION THREAD

**Discussion by RazaZaidi2802** [JSBSim-Team/jsbsim#1004] [Support requests]
https://youtu.be/MUC7h6EJWcA In this video, the aircraft is spawned near level origin and acts completely normal, but when spawned a very large distance away from origin, it begins to skid towards world origin and controls are completely unresponsive. I think this might be due to the GeoReferencingSystem actor.
These are the settings in the actor. Does anyone know why this may be happening?

![289593557-ddfa74a7-9834-409b-8b51-b1c409aab846](https://github.com/JSBSim-Team/jsbsim/assets/58148776/807dc489-430d-4533-b598-73d7690adb40)

**Comment by gallonmate** [comment]
The system in the UE5 plugin is currently only setup for a round earth/planet. So the gravity, air properties, etc are all simulated following a curvature of the earth with a realistic atmosphere eventually bleeding off into a vacuum of outer space. If you look at your altitude as seen in the video:
![image](https://github.com/JSBSim-Team/jsbsim/assets/28702753/ecf2a3ce-48c9-432b-8724-8c014f2d3b26)
 Then you'll notice you're now 4 million feet above "sea level", which is 771 miles above the simulation's ground level!

Basically if you go too far left, right, up or down, from 0,0,0, eventually you leave the planet!

Simply changing the Planet Shape to Flat planet in the Geo-reference Plugin would solve this problem but the UE5 plugin needs a few math adjusts to convert the Flat planet coordinates correctly, (it currently reverses the direction of the aircraft simulation).

If you know some c++ and apply the correct coordinate math then that's all that's needed. Otherwise you'll have to try something else, like possibly in blueprints you can use SetWorldOriginLocation to manually offset the origin closer to your level's location.

**Comment by RazaZaidi2802** [comment]
Thank you so much for that explanation! I should be able to have someone on my team modify the georeferencing plugin accordingly

**Reply by gallonmate** [reply]
You're welcome. And to be clear, it's the JSBSim UE5 plugin that needs the changes to accept the Flat planet coordinates. The Geo-reference plugin is okay. 

Also if you do try the SetWorldOriginLocation, feel free to report back so that others can know to use that as a possible solution.

**Comment by RazaZaidi2802** [comment]
Thank you! I'll forward the correction! Also, I tried the SetWorldOriginLocation as the Aircraft's current location and it just teleported my aircraft to the World Origin instead

https://github.com/JSBSim-Team/jsbsim/assets/58148776/d2cc82c7-fc39-4129-bfaa-f2aa1422c246


![image](https://github.com/JSBSim-Team/jsbsim/assets/58148776/191c7537-6ed9-4e05-918c-20bab333cee0)
